File: Python/enzoniko-cs50-problems-2019-x-sentimental-cash/cash.py
# Importa função "get_float" da biblioteca cs50
from cs50 import get_float
# Variáveis
m = 0
atual = 25
# Enquanto o valor não for positivo, repete a pergunta
while True:
    valor = get_float("Change Owed? ")
    apagar = round(valor * 100)
    if valor > 0:
        break
# Enquanto o valor for positivo, faz os loops
while (valor > 0):
    if (atual <= apagar):
        apagar -= atual
        m += 1
    else:
        if (apagar == 0):
            break
        elif (atual == 25):
            atual = 10
        elif (atual == 10):
            atual = 5
        elif (atual == 5):
            atual = 1
# Printa a quantidade mínima de moedas
print(m)

File: C/enzoniko-cs50-problems-2019-x-cash/cash.c
//Bibliotecas:
#include <cs50.h>
#include <stdio.h>
#include <math.h>
//Função main:
int main(void)
{
    //variáveis:
    float valor = printf("Change Owed? ");
    scanf("%f", &valor);
    int m = 0;
    int atual = 25;
    int apagar = round(valor * 100);
    //verifica se o input é positivo:
    while (valor <= 0)
    {
        printf("Change Owed? ");
        scanf("%f", &valor);
        apagar = round(valor * 100);
    }
    //se for positvo faz as coisas abaixo:
    while (valor > 0)
    {
        //da a quantidade de moedas:
        if (atual <= apagar)
        {
            apagar -= atual;
            m += 1;
        }
        else
        {
            //faz os cálculos diminuindo a quantidade do dinheiro:
            if (apagar == 0)
            {
                break;
            }
                
            else if (atual == 25)
            {
                atual = 10; 
            }
                
            else if (atual == 10)
            {
                atual = 5; 
            }
                
            else if (atual == 5)
            {
                atual = 1;
            }
        } 
    }   
    //print da quantidade de moedas
    printf("%i\n", m); 
}
